src/Functions/custom/HelperFunctions.ps1

```diff
@@ -0,0 +1,670 @@
+    )
+
+    if ($InputObject.Name -and $InputObject.ResourceGroupName -and $InputObject.SubscriptionId)
+    {
+        $functionsIdentity = New-Object -TypeName Microsoft.Azure.PowerShell.Cmdlets.Functions.Models.FunctionsIdentity
+        $functionsIdentity.Name = $InputObject.Name
+        $functionsIdentity.SubscriptionId = $InputObject.SubscriptionId
+        $functionsIdentity.ResourceGroupName = $InputObject.ResourceGroupName
+
+        return $functionsIdentity
+    }
+    
+    return $null
+}
+
+function GetSkuName
+{
+    param
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $Sku
+    )
+
+    if (($Sku -eq "D1") -or ($Sku -eq "SHARED"))
+    {
+        return "SHARED"
+    }
+    elseif (($Sku -eq "B1") -or ($Sku -eq "B2") -or ($Sku -eq "B3") -or ($Sku -eq "BASIC"))
+    {
+        return "BASIC"
+    }
+    elseif (($Sku -eq "S1") -or ($Sku -eq "S2") -or ($Sku -eq "S3"))
+    {
+        return "STANDARD"
+    }
+    elseif (($Sku -eq "P1") -or ($Sku -eq "P2") -or ($Sku -eq "P3"))
+    {
+        return "PREMIUM"
+    }
+    elseif (($Sku -eq "P1V2") -or ($Sku -eq "P2V2") -or ($Sku -eq "P3V2"))
+    {
+        return "PREMIUMV2"
+    }
+    elseif (($Sku -eq "PC2") -or ($Sku -eq "PC3") -or ($Sku -eq "PC4"))
+    {
+        return "PremiumContainer"
+    }
+    elseif (($Sku -eq "EP1") -or ($Sku -eq "EP2") -or ($Sku -eq "EP3"))
+    {
+        return "ElasticPremium"
+    }
+    elseif (($Sku -eq "I1") -or ($Sku -eq "I2") -or ($Sku -eq "I3"))
+    {
+        return "Isolated"
+    }
+
+    $guidanceUrl = 'https://docs.microsoft.com/en-us/azure/azure-functions/functions-premium-plan#plan-and-sku-settings'
+
+    $errorMessage = "Invalid sku (pricing tier), please refer to '$guidanceUrl' for valid values."
+    $exception = [System.InvalidOperationException]::New($errorMessage)
+    ThrowTerminatingError -ErrorId "InvalidSkuPricingTier" `
+                          -ErrorMessage $errorMessage `
+                          -ErrorCategory ([System.Management.Automation.ErrorCategory]::InvalidOperation) `
+                          -Exception $exception
+}
+
+function ThrowTerminatingError
+{
+    param 
+    (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $ErrorId,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        $ErrorMessage,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [System.Management.Automation.ErrorCategory]
+        $ErrorCategory,
+
+        [Exception]
+        $Exception,
+
+        [object]
+        $TargetObject
+    )
+
+    if (-not $Exception)
+    {
+        $Exception = New-Object -TypeName System.Exception -ArgumentList $ErrorMessage
+    }
+
+    $errorRecord = New-Object -TypeName System.Management.Automation.ErrorRecord -ArgumentList ($Exception, $ErrorId, $ErrorCategory, $TargetObject)
```

Comment:
Not sure about the usage for this? Why not use powershell's PSCmdlet.ThrowTerminatingError method?

Reply:
PSCmdlet.ThrowTerminatingError does not stop execution. For example:

I create an error record, and then throw it using PSCmdlet.ThrowTerminatingError. As you can see, the write-host statement after ";" is executed. I will keep throw for now.
```
$guidanceUrl = 'https://docs.microsoft.com/en-us/azure/azure-functions/functions-premium-plan#plan-and-sku-settings'
$errorMessage = "Invalid sku (pricing tier), please refer to '$guidanceUrl' for valid values."
$exception = [System.InvalidOperationException]::New($errorMessage)
ThrowTerminatingError -ErrorId "InvalidSkuPricingTier" `
                          -ErrorMessage $errorMessage `
                           -ErrorCategory ([System.Management.Automation.ErrorCategory]::InvalidOperation) `
                           -Exception $exception ; Write-Host "hello" -fore Yellow

ThrowTerminatingError : Invalid sku (pricing tier), please refer to 'https://docs.microsoft.com/en-us/azure/azure-functions/functions-premium-plan#plan-and-sku-settings' for valid values.
At line:1 char:5
+     ThrowTerminatingError -ErrorId "InvalidSkuPricingTier" `
+     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : InvalidOperation: (:) [ThrowTerminatingError], InvalidOperationException
+ FullyQualifiedErrorId : InvalidSkuPricingTier,ThrowTerminatingError

hello